### ISSUE TITLE
Support example values

### ISSUE THREAD

**Issue by focusaurus** [bjrmatos/json-schema-sugar#3]
I'd like to be able to do `schema.example("spades")` and get a property `{example: "spades"}` on the final object. This is of course not part of JSON schema AFAIK but is part of OpenAPI, where it is used by tools such as swagger-ui to construct example objects in the documentation. I only tested a single JSON schema validator (ajv) but it had no issue with the presence of the `example` property.

**Comment by bjrmatos** [comment]
hi! do you mean something like this?:

```js
var schema = sugar.object().keys({
  name: sugar.string(),
  creditCard: sugar.number(),
  billingAddress: sugar.string()
}).example('spades');

schema.generate();
```

and get something like this?:

```js
{
  type: 'object',
  properties: {
    name: {
      type: 'string'
    },
    creditCard: {
      type: 'number'
    },
    billingAddress: {
      type: 'string'
    },
  },
  example: 'spades'
}
```

if that is the case we can have a new method in `mixed` type, something like `mixed.meta`, this function could have a signature like this: `mixed.meta(metadata: object): Schema`, this function will let you store custom properties in the schema (useful if you would like to have custom data that are not part of json schema standard). this function should allow to be called multiples times, and extends the internal metadata information.

usage: 

```js
var schema = sugar.object().keys({
  name: sugar.string(),
  creditCard: sugar.number(),
  billingAddress: sugar.string()
}).meta({
  example: 'spades',
  anyOtherProperty: 'custom value'
});

schema.generate();
```

you will get something like this:

```js
{
  type: 'object',
  properties: {
    name: {
      type: 'string'
    },
    creditCard: {
      type: 'number'
    },
    billingAddress: {
      type: 'string'
    },
  },
  example: 'spades',
  anyOtherProperty: 'custom value'
}
```

with multiple calls:

```js
var schema = sugar.object().keys({
  name: sugar.string(),
  creditCard: sugar.number(),
  billingAddress: sugar.string()
}).meta({
  example: 'spades'
}).meta({
  anyOtherProperty: 'custom value'
});

schema.generate();
```

```js
{
  type: 'object',
  properties: {
    name: {
      type: 'string'
    },
    creditCard: {
      type: 'number'
    },
    billingAddress: {
      type: 'string'
    },
  },
  example: 'spades',
  anyOtherProperty: 'custom value'
}
```

would you like to implement this in a PR? if not, let me find sometime to add this new method, thnks

**Comment by focusaurus** [comment]
The way it's used in openapi/swagger, you only need examples on scalars, and those are used to infer parent object examples. So for example:

```
sugar.object().keys({
  name: sugar.string().example("Bob"),
  age: sugar.number().integer().minimum(0).example(42),
  domHand: sugar.string().enum(["left", "right", "ambidextrous"]).example("left")  
})
```

Would generate this object

```
{
  type: "object",
  properties: {
     name: { type: "string", example: "Bob" },
     age: { type: "integer", minimum: 0, example: 42 },
     domHand: { type: "string", enum: [ "left", "right", "ambidextrous" ], example: "left" }
  }
}
```

From which swagger would infer the following example:

```
{
  name: "Bob",
  age: 42,
  domHand: "left"
}
```

So at least for my purposes I'd only need `.example` on the scalar types.

As for `.meta` if we want that, I'd rather name it `.extra` but I think it goes against the grain of this library if you then have to write out object literals. I'd almost rather just write the json schema literals in that case.

**Comment by bjrmatos** [comment]
and what about if `meta/extra` would let you do this too?:

```js
sugar.object().keys({
  name: sugar.string().meta("example", "Bob"),
  age: sugar.number().integer().minimum(0).meta("example", 42),
  domHand: sugar.string().enum(["left", "right", "ambidextrous"]).meta("example", "left")  
})
```

````
{
  type: "object",
  properties: {
     name: { type: "string", example: "Bob" },
     age: { type: "integer", minimum: 0, example: 42 },
     domHand: { type: "string", enum: [ "left", "right", "ambidextrous" ], example: "left" }
  }
}
```

**Comment by focusaurus** [comment]
Yeah that saves you a few punctuation chars so I'd be OK with that.

**Comment by bjrmatos** [comment]
Ok!, i will work on that over the weekend, thnks!

**Comment by bjrmatos** [comment]
implemented in `v1.1.0`
